test/integration/cds_integration_test.cc

```diff
@@ -222,6 +234,76 @@ TEST_P(CdsIntegrationTest, CdsClusterTeardownWhileConnecting) {
   EXPECT_LE(cx_counter->value(), 1);
 }
 
+// Test that TrafficStats lazyinit when configured.
+TEST_P(CdsIntegrationTest, TrafficStatsLazyInit) {
```

Comment:
this is a great family of tests. I wonder if it's worth making a test subclass for this and making sure we cover a few cases, maybe each in separate tests:

1a lazy-init off, stats appear as soon as clusters are created. and there is value continuity if we send a request to a cluster, create a new config for the cluster, and then drop the old config for the cluster.
1b same as above but we do create/destroy/create, and ensure the value is actually lost, as expected.
2a lazy-init on, same 2 tests,

I think this would be easier to read as separate tests, with helper functions to factor out common parts, rather than parameterizing all the cds tests on lazy-init (most won't care).

WDYT?